Mathlib/LinearAlgebra/CliffordAlgebra/Prod.lean

```diff
@@ -0,0 +1,141 @@
+import Mathlib.LinearAlgebra.TensorProduct.Graded.Internal
+import Mathlib.LinearAlgebra.QuadraticForm.Prod
+
+/-!
+# Clifford algebras of a direct sum of two vector spaces
+
+We show that the clifford algebra of a direct sum is the graded tensor product of the clifford
+algebras, as `CliffordAlgebra.equivProd`.
+
+## Main definitions:
+
+* `CliffordAlgebra.equivProd : CliffordAlgebra (Q₁.prod Q₂) ≃ₐ[R] (evenOdd Q₁ ᵍ⊗[R] evenOdd Q₂)`
+
+-/
+
+suppress_compilation
+
+variable {R M₁ M₂ : Type*}
+variable [CommRing R] [AddCommGroup M₁] [AddCommGroup M₂] [Module R M₁] [Module R M₂]
+
+variable (Q₁ : QuadraticForm R M₁) (Q₂ : QuadraticForm R M₂)
+
+open scoped TensorProduct
+
+namespace CliffordAlgebra
+
+theorem map_inl_mul_map_inr_of_mem_evenOdd {i₁ i₂ : ZMod 2}
+    (m₁ : CliffordAlgebra Q₁) (hm₁ : m₁ ∈ evenOdd Q₁ i₁)
+    (m₂ : CliffordAlgebra Q₂) (hm₂ : m₂ ∈ evenOdd Q₂ i₂) :
+    map (.inl Q₁ Q₂) m₁ * map (.inr Q₁ Q₂) m₂ =
+      (-1 : ℤˣ) ^ (i₂ * i₁) • (map (.inr Q₁ Q₂) m₂ * map (.inl Q₁ Q₂) m₁) := by
+  -- the strategy; for each variable, induct on powers of `ι`, then on the exponent of each
+  -- power.
+  induction hm₁ using Submodule.iSup_induction' with
+  | h0 => rw [map_zero, zero_mul, mul_zero, smul_zero]
+  | hadd _ _ _ _ ihx ihy => rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+  | hp i₁' m₁' hm₁ =>
+    obtain ⟨i₁n, rfl⟩ := i₁'
+    dsimp only at *
+    induction hm₁ using Submodule.pow_induction_on_left' with
+    | hr =>
+      rw [AlgHom.commutes, Nat.cast_zero, mul_zero, uzpow_zero, one_smul, Algebra.commutes]
+    | hadd _ _ _ _ _ ihx ihy =>
+      rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+    | hmul m₁ hm₁ i x₁ _hx₁ ih₁ =>
+      obtain ⟨v₁, rfl⟩ := hm₁
+      -- this is the first interesting goal
+      rw [map_mul, mul_assoc, ih₁, mul_smul_comm, map_apply_ι, Nat.cast_succ, mul_add_one,
+        uzpow_add, mul_smul, ← mul_assoc, ← mul_assoc, ← smul_mul_assoc ((-1) ^ i₂),
+        QuadraticForm.Isometry.inl_apply]
+      clear ih₁
+      congr 2
+      induction hm₂ using Submodule.iSup_induction' with
+      | h0 => rw [map_zero, zero_mul, mul_zero, smul_zero]
+      | hadd _ _ _ _ ihx ihy => rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+      | hp i₂' m₂' hm₂ =>
+        clear m₂
+        obtain ⟨i₂n, rfl⟩ := i₂'
+        dsimp only at *
+        induction hm₂ using Submodule.pow_induction_on_left' with
+        | hr =>
+          rw [AlgHom.commutes, Nat.cast_zero, uzpow_zero, one_smul, Algebra.commutes]
+        | hadd _ _ _ _ _ ihx ihy =>
+          rw [map_add, add_mul, mul_add, ihx, ihy, smul_add]
+        | hmul m₂ hm₂ i x₂ _hx₂ ih₂ =>
+          obtain ⟨v₂, rfl⟩ := hm₂
+          -- this is the second interesting goal
+          rw [map_mul, map_apply_ι, QuadraticForm.Isometry.inr_apply, Nat.cast_succ, ← mul_assoc,
+            ι_mul_ι_comm_of_isOrtho (.inl_inr _ _), neg_mul, mul_assoc, ih₂, mul_smul_comm,
+            ← mul_assoc, ← Units.neg_smul, uzpow_add, uzpow_one, mul_neg_one]
+
+
```

Comment:
I think it's probably worth singling out these corollaries:
```suggestion

variable (m₁ : CliffordAlgebra Q₁) (m₂ : CliffordAlgebra Q₂)

theorem Commute_map_inl_inr_of_mem_evenOdd_zero_left {i : ZMod 2}
     (hm₁ : m₁ ∈ evenOdd Q₁ 0) (hm₂ : m₂ ∈ evenOdd Q₂ i) :
    Commute (map (.inl Q₁ Q₂) m₁) (map (.inr Q₁ Q₂) m₂) :=
  show _ * _ = _ * _ from by simp [map_inl_mul_map_inr_of_mem_evenOdd _ _ _ hm₁ _ hm₂]

theorem Commute_map_inl_inr_of_mem_evenOdd_zero_right {i : ZMod 2}
    (hm₁ : m₁ ∈ evenOdd Q₁ i) (hm₂ : m₂ ∈ evenOdd Q₂ 0) :
    Commute (map (.inl Q₁ Q₂) m₁) (map (.inr Q₁ Q₂) m₂) :=
  show _ * _ = _ * _ from by simp [map_inl_mul_map_inr_of_mem_evenOdd _ _ _ hm₁ _ hm₂]

theorem map_inl_mul_map_inr_eq_neg_of_mem_evenOdd_one
    (hm₁ : m₁ ∈ evenOdd Q₁ 1) (hm₂ : m₂ ∈ evenOdd Q₂ 1) :
    (map (.inl Q₁ Q₂) m₁) * (map (.inr Q₁ Q₂) m₂) =
    - (map (.inr Q₁ Q₂) m₂) * (map (.inl Q₁ Q₂) m₁) := by
  simp [map_inl_mul_map_inr_of_mem_evenOdd _ _ _ hm₁ _ hm₂]

```

Reply:
I ended up generalizing the statement away from `inl` and `inr`, but added analogous versions of your lemmas.

Perhaps in future we needed a `GradedCommute` helper to handle this family of lemmas automatically...